AEG-31-DN

D-Nugent opened the pull request:
Completed all adjusted discussed in last meeting.
Remaining considerations:
- All drawable SVG's were moved to one reusable component with object literals used to return each attribute. For reusability, we may want to adjust the component so you can pass props that alter the position where the SVG begins drawing or perhaps the speed of the animation. Not needed for any of the current SVG's but may be for future SVG's.
- Debounce was moved into a new utilFunctions folder. I considered making it a customHook as well but the way I viewed it, we should only place functions in a customHook when they use lifecycle methods (i.e. state, effects etc) and in this instance debounce does not.

D-Nugent committed:
Initial commit (TSX,Basic Styling, Updates to Global):
- Styling up until end of 'Looking for an update completed'
- Page to be broken into components in next commit
- Added some comments to fonts.scss to improve readability
- Deleted the mixin for mobile as all default styling should auto-apply
 to mobile device styling
- Added current figma color theme
- Removed (commented out) the font-sizing on html as this is altering
 the root font throughout the document.
 - Slight tweak to HTML reset with box-sizing and inherit
 - Experimented with creating a dynamic mixin for margins

D-Nugent committed:
Updated name for 'responsiveMargin' and created 'responsivePadding'

D-Nugent committed:
Initial Design 95% complete:
- All elements, images styled aside from the scroll-tracker
- Page still needs to be broken into components
- Basic form state control to be added.

D-Nugent committed:
Merge branch 'master' of https://github.com/D-Nugent/aegis-client into AEG-31-DN

D-Nugent committed:
Start of SVG background animation work.
- Created custom hook for scroll tracking
- Began dividing up components
- Implemented first SVG background image.

D-Nugent committed:
Added additional homepage svg images

D-Nugent committed:
Final Commit For First PR:
- Slight fix applied to debounce in custom hook
- Home.tsx separated into components
- SVG animations switched from length on scroll to transitions

D-Nugent committed:
Restructuring of SVG's and useScrollTrack:
- Have moved all drawable SVG's into their own component.
- Have moved debounce from inside of useScrollTrack to the utilFunctions
folder.

D-Nugent committed:
Changed naming conventions for Home Page components, components are
now prefixed with 'Home' so that they are grouped together

JamesAKing commented:
Visual Bug: Spacing surrounding FAQ section. Client and User button are overlapping / close to the Contact Us text at smaller screen sizes (320px - 768px screen width) - see Jira ticket point 1 for screenshots.

JamesAKing commented:
For future: Consider a smaller Return to top button that is restricted to the padding / margin area so as to stop button overlaying text (See Jira ticket point 2 for screen shot). 
[Like this design](http://nelucebotari.com/) (view @ desktop screen size)

JamesAKing commented:
Some strange visual overlapping occurring on my laptop screen but not on my monitor at the same screen width. Appears to be linked to screen height. 
Potential Fix: min height use absolute units.

JamesAKing commented:
Getting a memory leak warning when going from Patron Portal to Home Page. 
Appears t be linked to the drawable SVG. Created a bug ticket for this to be looked. 
https://aegis-dev.atlassian.net/browse/AEG-47

JamesAKing commented on src/components/ReturnToTop/ReturnToTop.tsx:
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
Unused import - can this be removed?

JamesAKing commented on src/App.scss:
```diff
@@ -14,5 +15,23 @@ html, body, textarea, button {
 
 html {
-  @include text-regular;
+  // @include text-regular;
```
Can this be removed?

JamesAKing commented on src/App.scss:
```diff
@@ -14,5 +15,23 @@ html, body, textarea, button {
-  @include text-regular;
+  // @include text-regular;
+  box-sizing: border-box;
+  min-height: 100vh;
```
Should we change this to be absolute units to avoid potentially having a min-height that is too small (e.g. it is possible to have min-height 0 with this).

D-Nugent replied:
As discussed, this will set the overall viewport but not affect fixed sizing of child elements. i.e. if you have an element with 400px height, it will still be 400px. It will just allow relative sizing.

JamesAKing commented on src/components/DrawableSVG/DrawableSVG.scss:
```diff
@@ -0,0 +1,86 @@
+@use '../../styles/variables' as *;
+@use '../../styles/mixins' as *;
+
+.svg__lock{
```
svg classes share some code (e.g. display and position). Could we make this DRYer with an @extend?
Also could consider a BEM modifier: className-"svg svg--lock" if this is more descriptive?

D-Nugent replied:
Good note, ended up changing to:
```
.drawable-svg{
  ...
  &--lock{
    ...
  }
}
```

JamesAKing commented on src/components/DrawableSVG/DrawableSVG.tsx:
```diff
@@ -0,0 +1,74 @@
+
+export default function DrawableSVG({type}:DrawableSVGProps) {
+  const scrollPos = useScrollTrack();
+  useEffect(() => {
```
I don't think this is unmounting correctly and is causing a memory leak warning. 
Have set up as a separate bug ticket: https://aegis-dev.atlassian.net/browse/AEG-47

D-Nugent replied:
Will update on new ticket.

JamesAKing commented on src/pages/Home/Home.scss:
```diff
@@ -0,0 +1,236 @@
+  }
+}
+
+// Uncertain if we'll end up using this search bar elsewhere in the app, if we do
```
Brainstorm best way to log these? 
I'm concerned comments in code are easily lost in the future.

D-Nugent replied:
To be reviewed at a later date. For this specific case, as it was already being used in two locations  it has been moved to global styles next to the button.

JamesAKing commented on src/pages/Home/Home.scss:
```diff
@@ -0,0 +1,236 @@
+  background-position: center;
+  background-repeat: no-repeat;
+  clip-path: url(#svgPathFaq);
+  flex: 1.25;
```
Where we are only passing one variable, is it best to avoid using shorthand? 
Concerned that this is not explicit.

D-Nugent replied:
As discussed, agreed.

D-Nugent replied:
Just realized a misunderstanding on my part. I had thought of flex-shrink as the second value but the second value is flex-shrink _or_ flex-basis.  flex: 1.25 is the equivalent of:
```
flex-grow: 1.25;
flex-basis: 0;
```
As such the shorthand is a bit more concise here but we could also do:
```
flex: 1.25 0;
```

JamesAKing commented on src/styles/_mixins.scss:
```diff
@@ -16,4 +16,222 @@
+// At desktop: margin: 1rem
+
+// Example 2:
+// input: @include margin(m,bottom,left)
```
Alternative format for comments: 
const sum(a, b) => {
  // input: a and b, required, integers above 0
  // output: integer, the sum of a and b
};

JamesAKing replied:
Also consider typescript

D-Nugent replied:
Agreed, will switch to typescript block-style comments for future annotations in both js and scss

JamesAKing reviewed (changes requested):
See code for details

D-Nugent committed:
Fixed overlapping of elements on small page height
- For this, simply assigned min-height on each component.

D-Nugent commented:
> For future: Consider a smaller Return to top button that is restricted to the padding / margin area so as to stop button overlaying text (See Jira ticket point 2 for screen shot).
> [Like this design](http://nelucebotari.com/) (view @ desktop screen size)

On the fence about this one. Agree in concept about not having it cover text, however, from an accessibility standpoint I'm not sure how clear it is that this is a return to top button. I don't mind the idea of just changing it to an icon but perhaps we make it so that it is just an up arrow and then when you hover over it it expands to it's current appearance like below:
![image](https://user-images.githubusercontent.com/73457386/131398473-afd1ecb8-8fd9-4c29-8d3a-361c6e495297.png)

D-Nugent committed:
Multiple:
- Fixed memory leak in useScrollTrack
- Added documentation comments to all functions and components with
functionality
- Restructured some styling (moved to global)
